c_plus_plus/src/view/views/local_conference_view.cpp

```diff
@@ -1,74 +1,63 @@
 #include "local_conference_view.h"
 #include "ui_local_conference_view.h"
+#include "model/settings/settings_constants.h"
 
-#include <QCamera>
-#include <QCameraInfo>
 #include <QCameraViewfinder>
-#include <QListWidgetItem>
+#include <QUrl>
 
 namespace View
 {
 
-LocalConferenceView::LocalConferenceView(QWidget *parent)
+LocalConferenceView::LocalConferenceView(Model::ISettings& settings, QWidget *parent)
     : AbstractView("Local Conference", parent)
     , m_ui(new Ui::LocalConferenceView)
-    , m_camera(new QCamera(getCameraInfo()))
+    , m_recorder(new Model::Recorder(getCameraDevice(), this))
+    , m_settings(&settings)
     , m_cameraViewfinder(new QCameraViewfinder(this))
 {
     m_ui->setupUi(this);
     m_ui->virtualCameraLayout->addWidget(m_cameraViewfinder);
 
-    m_camera->setViewfinder(m_cameraViewfinder);
-
+    m_recorder->setCameraViewfinder(m_cameraViewfinder);
     m_cameraViewfinder->show();
 
     connect(m_ui->btnStartStopRecord, &QAbstractButton::clicked, [=]{ changeRecordButtonState(); });
 }
 
-QCameraInfo LocalConferenceView::getCameraInfo()
+void LocalConferenceView::changeRecordButtonState()
 {
-    QCameraInfo cameraInfo;
-    QList<QCameraInfo> cameras = QCameraInfo::availableCameras();
-    if(cameras.length() == 1)
+    m_recordButtonState = !m_recordButtonState;
+
+    if(m_recordButtonState)
     {
-        cameraInfo = QCameraInfo::defaultCamera();
+        m_ui->btnStartStopRecord->setText("Stop recording");
+        m_recorder->start(getOutputPath());
     }
     else
     {
-        foreach (const QCameraInfo &camInfo, cameras)
-        {
-            if (camInfo.deviceName() == "/dev/video1")   //TODO: get device from config file
-                cameraInfo = camInfo;
-        }
+        m_ui->btnStartStopRecord->setText("Start recording");
+        m_recorder->stop();
```

Comment:
It's a correct logic, but I would rather implement a state pattern like in the webview page. :)